R/exercise.R

```diff
@@ -789,6 +792,27 @@ merge_options <- function(preserved_opts, inherited_opts, static_opts = list())
   opts[!grepl("^exercise", names(opts))]
 }
 
+local_restore_options <- function(.local_envir = parent.frame()) {
+  options <- options()
+  withr::defer(restore_options(options), envir = .local_envir)
+
+  envvars <- Sys.getenv()
+  withr::defer(restore_envvars(envvars), envir = .local_envir)
+}
```

Comment:
Let's turn this into two methods to not overload the name / functionality

```suggestion
}

local_restore_envvars <- function() {
  envvars <- Sys.getenv()
  withr::defer(restore_envvars(envvars), envir = .local_envir)
}
```

Reply:
If you'd like to make a wrapper around these two functions, I'm ok with it.